empty frames with insertable-streams/audio-processing example

# Please read first!
Please use [discuss-webrtc](https://groups.google.com/forum/#!forum/discuss-webrtc) for general technical discussions and questions.

- [x] I have provided steps to reproduce
- [x] I have provided browser name and version
- [x] I have provided a link to the sample here or a modified version thereof

**Note: If the checkboxes above are not checked (which you do after the issue is posted), the issue will be closed.**

## Browser affected
Chrome 94.0.4606.54 (Official Build) (64-bit) on linux

## Description
When running th [insertable-streams/audio-processing example](https://github.com/webrtc/samples/tree/gh-pages/src/content/insertable-streams/audio-processing) on local machine: the function returned by `lowPassFilter()` fails with the following log:
```
main.js:126 Error from stream transform: TypeError: Cannot read properties of undefined (reading 'sampleRate')
    at Object.transform (main.js:75)
```
When I look into it I discover that the `frame` argument appears to be an empty object `{}` which makes `frame.buffer` undefined and thus `buffer.frame.sampleRate` property cannot be read.

Thinking maybe initial frames could be empty but eventually I would start receiving valid frames I added some sanity checking to bypass empty frames however valid frames never showed up.

## Steps to reproduce
Run the `insertable-streams/audio-processing` example.
Open console and see error in logs.

## Expected results
The example is supposed to playback mic audio with a lowpass filter applied.

## Actual results
The example logs an error and no audio plays out the speakers.

--- a/src/content/insertable-streams/audio-processing/index.html
+++ b/src/content/insertable-streams/audio-processing/index.html
@@ -56,9 +56,7 @@
 
     <p>
         <b>Note</b>: This sample is using an experimental API that has not yet been standardized. As
-        of 2021-02-09, this API is available in Chrome M90 if the experimental code is enabled on
-        the command line with
-        <code>--enable-blink-features=WebCodecs,MediaStreamInsertableStreams</code>.
+        of 2021-09-29, this API is available in Chrome M94.
     </p>
     <a href="https://github.com/webrtc/samples/tree/gh-pages/src/content/insertable-streams/audio-processing"
        title="View source for this page on GitHub" id="viewSource">View source on GitHub</a>

--- a/src/content/insertable-streams/audio-processing/js/main.js
+++ b/src/content/insertable-streams/audio-processing/js/main.js
@@ -69,18 +69,22 @@ const constraints = window.constraints = {
 
 // Returns a low-pass transform function for use with TransformStream.
 function lowPassFilter() {
+  const format = 'f32-planar';
   let lastValuePerChannel = undefined;
-  return (frame, controller) => {
+  return (data, controller) => {
     const rc = 1.0 / (cutoff * 2 * Math.PI);
-    const dt = 1.0 / frame.buffer.sampleRate;
+    const dt = 1.0 / data.sampleRate;
     const alpha = dt / (rc + dt);
-    const nChannels = frame.buffer.numberOfChannels;
+    const nChannels = data.numberOfChannels;
     if (!lastValuePerChannel) {
       console.log(`Audio stream has ${nChannels} channels.`);
       lastValuePerChannel = Array(nChannels).fill(0);
     }
+    const buffer = new Float32Array(data.numberOfFrames * nChannels);
     for (let c = 0; c < nChannels; c++) {
-      const samples = frame.buffer.getChannelData(c);
+      const offset = data.numberOfFrames * c;
+      const samples = buffer.subarray(offset, offset + data.numberOfFrames);
+      data.copyTo(samples, {planeIndex: c, format});
       let lastValue = lastValuePerChannel[c];
 
       // Apply low-pass filter to samples.
@@ -89,10 +93,9 @@ function lowPassFilter() {
         samples[i] = lastValue;
       }
 
-      frame.buffer.copyToChannel(samples, c);
       lastValuePerChannel[c] = lastValue;
     }
-    controller.enqueue(frame);
+    controller.enqueue(new AudioData({format, sampleRate: data.sampleRate, numberOfFrames: data.numberOfFrames, numberOfChannels: nChannels, timestamp: data.timestamp, data: buffer}));
   };
 }
 

--- a/src/content/insertable-streams/audio-processing/js/main.js
+++ b/src/content/insertable-streams/audio-processing/js/main.js
@@ -8,7 +8,7 @@
 
 'use strict';
 
-/* global MediaStreamTrackProcessor, MediaStreamTrackGenerator */
+/* global MediaStreamTrackProcessor, MediaStreamTrackGenerator, AudioData */
 if (typeof MediaStreamTrackProcessor === 'undefined' ||
     typeof MediaStreamTrackGenerator === 'undefined') {
   alert(
@@ -24,6 +24,12 @@ try {
   alert(
       'Your browser does not support insertable audio streams. See the note ' +
         'at the bottom of the page.');
+}
+
+if (typeof AudioData === 'undefined') {
+  alert(
+      'Your browser does not support WebCodecs. See the note at the bottom ' +
+      'of the page.');
 }
 
 // Put variables in global scope to make them available to the browser console.
@@ -95,7 +101,14 @@ function lowPassFilter() {
 
       lastValuePerChannel[c] = lastValue;
     }
-    controller.enqueue(new AudioData({format, sampleRate: data.sampleRate, numberOfFrames: data.numberOfFrames, numberOfChannels: nChannels, timestamp: data.timestamp, data: buffer}));
+    controller.enqueue(new AudioData({
+      format,
+      sampleRate: data.sampleRate,
+      numberOfFrames: data.numberOfFrames,
+      numberOfChannels: nChannels,
+      timestamp: data.timestamp,
+      data: buffer
+    }));
   };
 }
 
@@ -104,7 +117,9 @@ async function start() {
   try {
     stream = await navigator.mediaDevices.getUserMedia(constraints);
   } catch (error) {
-    const errorMessage = 'navigator.MediaDevices.getUserMedia error: ' + error.message + ' ' + error.name;
+    const errorMessage =
+          'navigator.MediaDevices.getUserMedia error: ' + error.message + ' ' +
+          error.name;
     document.getElementById('errorMsg').innerText = errorMessage;
     console.log(errorMessage);
   }

